warehouse/email/__init__.py

```diff
@@ -56,8 +71,16 @@ def _send_email_to_user(request, user, msg, *, email=None, allow_unverified=Fals
     if email is None or not (email.verified or allow_unverified):
         return
 
+    # We should only store/display IP address of an 'email sent' event if the user
+    # who triggered the email event is the one who receives the email. Else display
+    # 'Redacted' to prevent user privacy concerns
+    user_email = request.db.query(Email).filter(Email.email == email.email).one()
+    ip_address = (
+        request.remote_addr if user_email.user_id == request.user.id else "Redacted"
+    )
```

Comment:
I think instead of the `ip_address` literally becoming `"Redacted"` here, we should set something like `additional.redact_ip` and just change how we display it in the UI with something like:

```
{{ "Redacted" if event.additional.redact_ip else event.ip_address }}
```
So this would become something like:

```suggestion
    user_email = request.db.query(Email).filter(Email.email == email.email).one()
    redact_ip = user_email.user_id != request.user.id
```

Reply:
My thought process here was that if we store the IP in the database and redact it in the UI, then there is a possibility that a user can manipulate the client and still extract the IP. Hence, rather not have that data to begin with. However, I'm not sure how feasible extracting the data would be from the client and if we need to worry about that.

Reply:
I think we still want the data available for administrators. I don't think it would be possible to extract the IP.